lib/panda/error_handler.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'faraday'
+module FaradayMiddleware
+  class PandaErrorHandler < Faraday::Middleware
+    def on_complete(env)
+      raise Panda::APIError.new(env.status, env.body) if env[:status] != 200
+
+      parsed_body = JSON.parse(env.body)
+      raise_error(parsed_body, env.status, env.body)
+    end
+
+    private
+
+    def raise_error(parsed_body, status, body)
```

Comment:
The naming is unclear. `raise_error` doesn't reflect that this method won't raise anything if the body code field is not 4xxxx. Let's call it `check_and_raise!`. Plus it's better to parse body inside this method. That way it will only accept env and not 3 arguments.